pkg/process/dockerproxy/filter.go

```diff
@@ -0,0 +1,164 @@
+// +build !windows
+
+package dockerproxy
+
+import (
+	"strconv"
+	"strings"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/gopsutil/process"
+)
+
+// Filter keeps track of every docker-proxy instance and filters network traffic going through them
+type Filter struct {
+	proxyByTarget map[target]*proxy
+	// This "secondary index" is used only during the proxy IP discovery process
+	proxyByPID map[int32]*proxy
+}
+
+type target struct {
+	ip    string
+	port  int32
+	proto model.ConnectionType
+}
+
+type proxy struct {
+	pid    int32
+	ip     string
+	target target
+}
+
+// NewFilter instantiates a new filter loaded with docker-proxy instance information
+func NewFilter() *Filter {
+	filter := new(Filter)
+	if procs, err := process.AllProcesses(); err == nil {
+		filter.LoadProxies(procs)
+	} else {
+		log.Errorf("error initiating proxy filter: %s", err)
+	}
+
+	return filter
+}
+
+// LoadProxies by inspecting processes information
+func (f *Filter) LoadProxies(procs map[int32]*process.FilledProcess) {
+	f.proxyByPID = make(map[int32]*proxy)
+	f.proxyByTarget = make(map[target]*proxy)
+
+	for _, p := range procs {
+		proxy := extractProxyTarget(p)
+		if proxy == nil {
+			continue
+		}
+
+		log.Debugf("detected docker-proxy with pid=%d target.ip=%s target.port=%d target.proto=%s",
+			proxy.pid,
+			proxy.target.ip,
+			proxy.target.port,
+			proxy.target.proto,
+		)
+
+		// Add proxy to cache
+		f.proxyByPID[proxy.pid] = proxy
+		f.proxyByTarget[proxy.target] = proxy
+	}
+}
+
+// Filter all connections that have a docker-proxy at one end
+func (f *Filter) Filter(payload *model.Connections) {
+	if len(f.proxyByPID) == 0 {
+		return
+	}
+
+	// Discover proxy IPs
+	// TODO: we can probably discard the whole logic below if we determine that each proxy
+	// instance will be always bound to the docker0 IP
```

Comment:
That would be nice, do we know how we'll confirm this?

Reply:
I'll definitely do some digging. But I think we should keep the current approach for this next release.